scripts/reboot

```diff
@@ -178,6 +190,8 @@ if [ -x ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE} ]; then
     ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE}
 fi
 
+stop_pmon_service
```

Comment:
I understand you want to stop this at the last possible moment, but does it just make things simpler to add the logic to the existing `stop_sonic_services()` function?

Reply:
If there is any platform API is introduced/invoked in between after stopping PMON, it will fail if PMON is shutdown early.

Reply:
I don't see a problem moving stop_sonic_service to here.

Reply:
@aravindthoram can we consolidate the change in stop_sonic_services and move the call to there?

Reply:
@yxieca Will commit the new changes soon.